contracts/mock-intents/src/lib.rs

```diff
@@ -1,21 +1,41 @@
+use near_sdk::log;
 use near_sdk::store::LookupMap;
 use near_sdk::{
-    assert_one_yocto, env, near, AccountId, BorshStorageKey, PanicOnDefault, PublicKey,
+    assert_one_yocto, env, ext_contract, json_types::U128, near, AccountId, BorshStorageKey,
+    NearToken, PanicOnDefault, PromiseError, PromiseOrValue, PublicKey,
 };
 use std::collections::HashSet;
 
+mod token;
 mod token_receiver;
 
+type Balance = u128;
+
+use crate::token::TokenId;
+
+#[allow(dead_code)]
+#[ext_contract(ext_ft)]
+trait FungibleTokenContract {
+    fn ft_transfer(
+        receiver_id: AccountId,
+        amount: U128,
+        memo: Option<String>,
+    ) -> PromiseOrValue<U128>;
+}
+
 #[derive(PanicOnDefault)]
 #[near(contract_state)]
 pub struct Contract {
     public_keys: LookupMap<AccountId, HashSet<PublicKey>>,
+    mt_balances: LookupMap<TokenId, LookupMap<AccountId, Balance>>,
```

Comment:
I believe we can get rid of the underlying `LookupMap` by using a tulip as a key:

```suggestion
    mt_balances: LookupMap<(TokenId, AccountId), Balance>,
```
?